### PR TITLE
User: use update function for password updates

### PR DESCRIPTION
**Pull request by kalleep**
**What is this feature?**
Similar to changes done in https://github.com/grafana/grafana/pull/86341 this pr makes it so that it is possible to to use `Update` to update password of a user.

I also did a bit of refactoring around update password handlers to reduce duplicated code. 

1. Extracted a function to check if user is external.

2. Move validation and hashing responsibility to service Update function. This makes sure that we validate and hash passwords in a single place instead of having to remember to do in each call site. It also aligns with what we do in Create function

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/user/userimpl/store.go [hunk 1]
```diff
@@ -64,6 +63,9 @@ func (ss *sqlStore) Insert(ctx context.Context, cmd *user.User) (int64, error) {
 			cmd.UID = util.GenerateShortUID()
 		}
 
+		cmd.Email = strings.ToLower(cmd.Email)
+		cmd.Login = strings.ToLower(cmd.Login)
```
Comment (kalleep) [inline pkg/services/user/userimpl/store.go:67]: This was done so that we can be sure that stored email and login is always lower-case

File: pkg/api/admin_users.go [hunk 2]
```diff
@@ -46,9 +44,6 @@ func (hs *HTTPServer) AdminCreateUser(c *contextmodel.ReqContext) response.Respo
 		return response.Error(http.StatusBadRequest, "bad request data", err)
 	}
 
-	form.Email = strings.TrimSpace(form.Email)
-	form.Login = strings.TrimSpace(form.Login)
```
Comment (kalleep) [inline pkg/api/admin_users.go:50]: Not needed anymore. Done in store instead

File: pkg/api/dtos/invite.go [hunk 3]
```diff
@@ -25,5 +25,5 @@ type CompleteInviteForm struct {
 	Name            string        `json:"name"`
 	Username        string        `json:"username"`
 	Password        user.Password `json:"password"`
-	ConfirmPassword user.Password `json:"confirmPassword"`
+	ConfirmPassword string        `json:"confirmPassword"`
```
Comment (linoman) [inline pkg/api/dtos/invite.go:28]: Why is this change needed?

Comment (linoman) [inline pkg/api/dtos/invite.go:28]: Why is this change needed?

**Review by linoman — APPROVED** [review]
Looks great. I left 1 question.

File: pkg/api/dtos/invite.go (hunk 3 above)
Comment (kalleep) [reply pkg/api/dtos/invite.go:28]: Will change back

Comment (kalleep) [reply pkg/api/dtos/invite.go:28]: Also that one seems to be unused

**Review by Jguer — APPROVED** [review]
